lib/mongo_mapper/plugins/dirty.rb

```diff
@@ -27,7 +27,11 @@ def clear_changes
         (block_given? ? yield : true).tap do |result|
           unless result == false #failed validation; nil is OK.
             @previously_changed = previous
-            changed_attributes.clear
+            if ::ActiveModel::VERSION::MAJOR > 4
+              changes_applied
+            else
+              changed_attributes.clear
+            end
```

Comment:
It seems we can use `changes_applied` even Rails 4.2

https://api.rubyonrails.org/v4.2.11/classes/ActiveModel/Dirty.html#method-i-changes_applied

```suggestion
            changes_applied
```

Reply:
Indeed, thank you~! :pray: